src/game_battler.cpp

```diff
@@ -696,12 +696,14 @@ std::vector<int16_t> Game_Battler::NextBattleTurn() {
 	for (size_t i = 0; i < states.size(); ++i) {
 		if (HasState(i + 1)) {
 			states[i] += 1;
-
-			if (states[i] >= Data::states[i].hold_turn) {
+			if (states[i] > Data::states[i].hold_turn + 1) {
 				if (Utils::ChanceOf(Data::states[i].auto_release_prob, 100)) {
 					healed_states.push_back(i + 1);
 					RemoveState(i + 1);
 				}
+				else {
+					states[i] -= Data::states[i].hold_turn;
```

Comment:
> 2.28: RM2000 + RM2003: If a battler is thrown a state again, the tur

Because the code before is
```cpp
states[i] += 1; 
if (states[i] > Data::states[i].hold_turn + 1) {
```

isn't this the same as ``states[i] = 1``?

Reply:
I don't think so.

`(s > ht + 1)` is equivalent to `(s - ht) > 1`. So `s >= 2` when that write happens.

Reply:
I have no idea if RPG2000 does this different but for RPG2003 I question the whole code except ```states[state_id - 1] = std::max<int> (1, states[state_id - 1]); ``` (which prevents turn count rejection when the same state is applied again). I used a Poison with 10 turns & 20% setting and I got it healt after turn 16... (according to this commit it should be again at 20?)